src/app/directives/has-permission/has-permission.directive.ts

```diff
@@ -0,0 +1,78 @@
+/** Angular Imports */
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+/**
+ * Has Permission Directive
+ */
+@Directive({
+  selector: '[mifosxHasPermission]'
+})
+export class HasPermissionDirective {
+
+  /** Describes Visibility */
+  private isVisible: boolean;
+  /** Flag if permission passed is negative */
+  private isNegation = false;
+  /** User Permissions */
+  private userPermissions: any[];
+  /** Key to store credentials in storage. */
+  private credentialsStorageKey = 'mifosXCredentials';
+
+  /**
+   * Extracts User Permissions from User Credentials
+   * @param {TemplateRef} templateRef Template Reference
+   * @param {ViewContainerRef} viewContainer View Container Reference
+   */
+  constructor(private templateRef: TemplateRef<any>,
+              private viewContainer: ViewContainerRef) {
+    const savedCredentials = JSON.parse(
+      sessionStorage.getItem(this.credentialsStorageKey) || localStorage.getItem(this.credentialsStorageKey)
+    );
+    this.userPermissions = savedCredentials.permissions;
+  }
+
+  /**
+   * Evaluates the condition to show template.
+   */
+  @Input()
+  set mifosxHasPermission(permission: any) {
+    if (typeof permission !== 'string') {
+      throw new Error('hasPermission value must be a string');
+    }
+    if (permission.charAt(0) === '!') {
+      this.isNegation = true;
+      permission = permission.slice(1);
+    }
+    if (this.hasPermission(permission) && !this.isNegation) {
```

Comment:
In community app we have two conditions here, did you check the specific use case for both possibilities? Why have we removed one of the conditions? https://github.com/openMF/community-app/blob/develop/app/scripts/directives/HasPermissionDirective.js

Reply:
if this is true second condition is also true, if this is false second condition is also false.

Reply:
I meant in community app there are two conditions: 
hasPermission && !notPermissionFlag || !hasPermission && notPermissionFlag
The conditions are different and cannot be true at the same time. Also, can you check if its being actually used anywhere? If yes we can add a comment here explaining its usage to avoid confusion otherwise we will just remove it if its not being used.

Reply:
Yes, it's not being used, I felt like removing it but wasn't sure back then.